### PR TITLE
Fixing Error State for Timeout/localhost Redirection

### PR DESCRIPTION
**Pull request by wayne-chen**
- Fix will allow when `ConnectionMonitor` times out, the `DomainHandler` will be set into an error state and will not retry connecting to the current domain URL (calling `AddressManager::refreshPreviousLookup`) every couple of seconds.
- Fix will also include the fix for MS#[18393](https://highfidelity.manuscript.com/f/cases/18393/404-Try-Again-portal-in-the-404-fails-to-take-the-user-to-the-intended-domain-after-correcting-the-issue)

## TEST PLAN
**For Testing Timeout**
- Launch Interface
- Try to connect to a domain that will time out connection (i.e. `hifi://localhost` if there is no sandbox open)
- Get redirected to 404 serverless domain
- Click `Settings > Developer Menu`
- Open `Developer > Console`
- Do not slowly change z-position for your avatar (checkable via typing `JSON.stringify(MyAvatar.position)` into the console every couple of seconds)
- Do not reconnect to the 404 serverless domain every couple of seconds

**For Testing localhost**
- Launch Interface
- Enter localhost
- Remove connect permissions for everyone but yourself
- Sign out of account
- End up in 404
- Log into account
- Try Again

### Note
When in the redirect domain, the window title should contain "`serverless: file:///~/serverless/redirect.json (ERROR CONNECTING)`"

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12939/HighFidelity-Beta-PR13984-3fe3f67.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12939/HighFidelity-Beta-PR13984-3fe3f67.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12939/HighFidelity-Beta-Interface-PR13984-3fe3f67.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12939/HighFidelity-Beta-PR13984-3fe3f67.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12939/HighFidelity-Beta-Interface-PR13984-3fe3f67.exe

File: libraries/networking/src/DomainHandler.cpp [hunk 1]
```diff
@@ -55,6 +55,9 @@ DomainHandler::DomainHandler(QObject* parent) :
 
     // stop the refresh timer if we connect to a domain
     connect(this, &DomainHandler::connectedToDomain, &_apiRefreshTimer, &QTimer::stop);
+
+    // stop the refresh timer if we connect to a domain
```
Comment (birarda) [inline libraries/networking/src/DomainHandler.cpp]: This comment is incorrect

File: libraries/networking/src/DomainHandler.cpp [hunk 2]
```diff
@@ -99,19 +102,21 @@ void DomainHandler::softReset() {
...
-    _isInErrorState = false;
     _connectionDenialsSinceKeypairRegen = 0;
     _checkInPacketsSinceLastReply = 0;
 
     // cancel the failure timeout for any pending requests for settings
     QMetaObject::invokeMethod(&_settingsTimer, "stop");
 
     // restart the API refresh timer in case we fail to connect and need to refresh information
-    QMetaObject::invokeMethod(&_apiRefreshTimer, "start");
+    if (!_isInErrorState)
```
Comment (birarda) [inline libraries/networking/src/DomainHandler.cpp]: Per the coding standard, the body of the conditional should be wrapped in`{ }`

File: libraries/networking/src/DomainHandler.cpp [hunk 3]
```diff
@@ -99,19 +102,21 @@ void DomainHandler::softReset() {
...
+        QMetaObject::invokeMethod(&_apiRefreshTimer, "start");
+    _isInErrorState = false;
 }
 
 void DomainHandler::hardReset() {
-    emit resetting();
+    if (!_isInErrorState)
+        emit resetting();
```
Comment (birarda) [inline libraries/networking/src/DomainHandler.cpp]: Per the coding standard, the body of the conditional should be wrapped in `{ }`

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12946/HighFidelity-Beta-PR13984-ad23367.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12946/HighFidelity-Beta-PR13984-ad23367.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12946/HighFidelity-Beta-Interface-PR13984-ad23367.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12946/HighFidelity-Beta-PR13984-ad23367.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12946/HighFidelity-Beta-Interface-PR13984-ad23367.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12950/HighFidelity-Beta-PR13984-5902699.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12950/HighFidelity-Beta-PR13984-5902699.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12950/HighFidelity-Beta-Interface-PR13984-5902699.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12950/HighFidelity-Beta-PR13984-5902699.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12950/HighFidelity-Beta-Interface-PR13984-5902699.exe

**Comment by qlilaseguinolaza** [comment]
Currently testing

File: libraries/networking/src/AddressManager.cpp [hunk 4]
```diff
@@ -269,6 +269,7 @@ bool AddressManager::handleUrl(const QUrl& lookupUrl, LookupTrigger trigger) {
                 // save the last visited domain URL.
                 _lastVisitedURL = lookupUrl;
 
+                auto lastVisitedURLStr = lookupUrl.toString().toStdString();
```
Comment (birarda) [inline libraries/networking/src/AddressManager.cpp]: Leftover debug variable

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12959/HighFidelity-Beta-PR13984-380f24b.apk

**Comment by JeffClinton** [comment]
Currently testing (once build is ready)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12959/HighFidelity-Beta-PR13984-380f24b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12959/HighFidelity-Beta-Interface-PR13984-380f24b.exe

**Comment by JeffClinton** [comment]
FAIL @conklin94122 

Attempting to enter the localhost when the server is stopped will take you to the 404.  Colliding with the Try Again portal fails to reload the 404.  You hang on the interstitial until using the abort button to escape.

Steps to repro:
1. Launch interface
2. Stop server on the localhost
3. Attempt to go to localhost
4. Collide with the Try Again button

interstitial
[ac-20180913.12.42.27.533-stdout.txt](https://github.com/highfidelity/hifi/files/2380817/ac-20180913.12.42.27.533-stdout.txt)

**Comment by JeffClinton** [comment]
Currently testing (once built)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12968/HighFidelity-Beta-PR13984-647c4e1.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12968/HighFidelity-Beta-PR13984-647c4e1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12968/HighFidelity-Beta-Interface-PR13984-647c4e1.exe

**Comment by JeffClinton** [comment]
PASS @Conklin94122

**Pull request merged**